src/utlist.h

```diff
@@ -349,6 +349,26 @@ do {
   }                                                                                            \
 } while (0)
 
+#define LL_INSERT_INORDER(head,add,cmp)                                                        \
+  do {                                                                                         \
+    LDECLTYPE(head) _pos;                                                                      \
```

Comment:
@Quuxplusone 
I resolved the head == NULL possibility in the INORDER macros and made the other changes that you advocated for.  I was not able to generate variants of the macros that would work with `NO_DECLTYPE` compilers - the only solution that I see would be passing a dummy node to the macros as an argument.

Please let me know if there are further changes I could make to improve this contribution!